### PR TITLE
Use rosgraph_msgs/Clock for /clock topic.

### PR DESCRIPTION
**Pull request by ernestmc**
Addresses issue #473 
Use `rosgraph_msgs/Clock` message for `/clock` topic instead of `builtin_interfaces/Time` to be compatible with ROS1.
This is needed for the `ros1_bridge` to effectively map `/clock` message when using Gazebo.

connects to #473

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
@ernestmc Thanks for the patch! We assumed this is ready for review so we labeled it as is (please comment here if this is not the case)
Tip: We also modified the description to connect to the relevant issue so that they are displayed together on the [waffle board](https://waffle.io/ros2/ros2)

**Comment by ernestmc** [comment]
@mikaelarguedas this is ready for review. I've tested and it works. Awaiting you comments.

**Comment by mikaelarguedas** [comment]
@ernestmc It looks like this PR makes multiple rclcpp tests fail.
Can you confirm if you have the same failures on your side ? (you can run the tests of a package by using `ament test --only <PKG_NAME>`)

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4438/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1380/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3627/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4498/)

**Comment by ernestmc** [comment]
Forgot to try the tests. I'll look at them and let you know.

**Comment by ernestmc** [comment]
@mikaelarguedas I've updated the test cases and now they are passing.
```
100% tests passed, 0 tests failed out of 27

Label Time Summary:
copyright     =   0.77 sec (1 test)
cppcheck      =   0.87 sec (1 test)
cpplint       =   5.72 sec (1 test)
gmock         =   0.23 sec (1 test)
gtest         =   7.29 sec (21 tests)
lint_cmake    =   0.70 sec (1 test)
linter        =  10.35 sec (5 tests)
uncrustify    =   2.28 sec (1 test)
```

**Comment by mikaelarguedas** [comment]
thanks @ernestmc 

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4481/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1411/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3670/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4556/)

**Comment by mikaelarguedas** [comment]
@ernestmc FYI: I rebased this branch on top of master and force pushed it

The reason: 
Our ci is unable to merge branches from forks at the moment and severals tests are failing due to the outdated state of this branch.

In the future we recommend, when possible, opening PRs from branches hosted on the ros2 repositories this way CI will make sure to merge your branch into master before building.

New round of CI on rebased branch:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4486/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1412/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3675/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4561/)

**Comment by ernestmc** [comment]
Thanks @mikaelarguedas . I've rebased my branch to keep it updated. I didn't know if I could push a new branch directly into the repo.

**Comment by mikaelarguedas** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4490/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1414/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3677/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4563/)

File: rclcpp/test/test_time_source.cpp [hunk 1]
```diff
@@ -93,18 +93,17 @@ TEST_F(TestTimeSource, clock) {
...
       break;  // Break for ctrl-c
     }
-    auto msg = std::make_shared<builtin_interfaces::msg::Time>();
-    msg->sec = i;
-    msg->nanosec = 1000;
+    auto msg = std::make_shared<rosgraph_msgs::msg::Clock>();
+    msg->clock.sec = i;
+    msg->clock.nanosec = 1000;
     clock_pub->publish(msg);
-    // std::cout << "Publishing: '" << msg->sec << ".000000" << msg->nanosec << "'" << std::endl;
```
Comment (mikaelarguedas) [inline rclcpp/test/test_time_source.cpp]: @tfoote did you want to keep these for debugging?

Comment (ernestmc) [reply rclcpp/test/test_time_source.cpp]: If we decide to keep those lines I need to update them to the new message type.

Comment (tfoote) [reply rclcpp/test/test_time_source.cpp]: No we don't need to keep them.

**Comment by mikaelarguedas** [comment]
thanks @ernestmc for the contribution and @tfoote for the second round of review, merging..

**Pull request merged**